.checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,140 @@
+            <property name="format" value="(^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$)|(^log$)"/>
+        </module>
+        <module name="LocalVariableName"/>
+        <module name="LocalFinalVariableName"/>
+        <module name="MemberName"/>
+        <module name="ClassTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="MethodTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="InterfaceTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="PackageName"/>
+        <module name="ParameterName"/>
+        <module name="StaticVariableName"/>
+        <module name="TypeName"/>
+        <module name="AvoidStarImport"/>
+
+        <!-- dependencies -->
+        <!--<module name="ImportControl">
+            <property name="file" value="${importControlFile}"/>
+        </module>-->
+
+        <!-- whitespace -->
+        <module name="GenericWhitespace"/>
+        <module name="NoWhitespaceBefore"/>
+        <module name="WhitespaceAfter" />
+        <module name="NoWhitespaceAfter"/>
+        <module name="WhitespaceAround">
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+        </module>
+        <module name="Indentation"/>
+        <module name="MethodParamPad"/>
+        <module name="ParenPad"/>
+        <module name="TypecastParenPad"/>
+
+        <!-- locale-sensitive methods should specify locale -->
+        <module name="Regexp">
+            <property name="format" value="\.to(Lower|Upper)Case\(\)"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="ignoreComments" value="true"/>
+        </module>
+
+        <!-- code quality -->
+        <module name="MethodLength"/>
+        <module name="ParameterNumber">
+            <!-- default is 8 -->
+            <property name="max" value="13"/>
+        </module>
+        <module name="ClassDataAbstractionCoupling">
+            <!-- default is 7 -->
+            <property name="max" value="20"/>
+        </module>
+        <module name="BooleanExpressionComplexity">
+            <!-- default is 3 -->
+            <property name="max" value="5"/>
+        </module>
+
+        <module name="ClassFanOutComplexity">
+            <!-- default is 20 -->
+            <property name="max" value="44"/>
+        </module>
+        <module name="CyclomaticComplexity">
+            <!-- default is 10-->
+            <property name="max" value="19"/>
+        </module>
+        <module name="JavaNCSS">
+            <!-- default is 50 -->
+            <property name="methodMaximum" value="100"/>
+        </module>
+        <module name="NPathComplexity">
+            <!-- default is 200 -->
+            <property name="max" value="5832"/>
+        </module>
```

Comment:
I am not sure we really need all the above "overrides" for the module above (i.e. NPathComplexity with 5832 instead of default 200). I mean, if this file was got from operator, the code there is fairly complex and to suppress warnings we had to increase these values. I would expect the client code is much simpler so less, maybe just the default, values are enough?

Reply:
Yeah, unfortunately, we screwed some of these numbers in other repos instead of using annotations to override where needed.

Reply:
Agree but let's try to not screw up this shiny new checkstyle as well. I would love having @OwenCorrigan76 to start by leaving defaults and see if the checkstyle complains.

Reply:
@ppatierno @scholzj  I'm just reading up on `Cyclomatic Complexity` and `NPath Complexity` now, as I had no idea what either was before now. Likewise with  `BooleanExpressionComplexity` and all of the other `code quality` modules. Do you suggest I remove some or all of the `code quality` modules and see if the defaults handle the checks without error?

Reply:
I went with the default values and ran into the following errors [1]. So now I know what to set those values to and leave the others as defaults.
[1]
` /home/owencorrigan/Documents/Work_Repos/client-examples/java/http/java-http-consumer/src/main/java/HttpConsumerConfig.java:27:13: More than 7 parameters (found 9). [ParameterNumber]
[ERROR] /home/owencorrigan/Documents/Work_Repos/client-examples/java/http/java-http-consumer/src/main/java/HttpConsumerConfig.java:40:5: NPath Complexity is 256 (max allowed is 200). [NPathComplexity]
`

Reply:
Or you can use annotations to override only where needed as Jakub suggested.

Reply:
@OwenCorrigan76 I saw you just changed the values in the checkstyle configuration properties. What Jakub was suggesting was about using the `@SuppressWarnings` annotation on the method/class where the warning was raised.
For example, using something like this:

```java
@SuppressWarnings("checkstyle:NPathComplexity")
```

instead of changing those values you can suppress the warning where needed.

Reply:
How many warnings do you get? One option is to use the annotations to suppress them in this PR and then get back to it and improve the code where it makes sense. Without seeing where it is complaining, it is a bit hard to say what seems more like a wrong limit in checkstyle and what is a _bad_ code.

Reply:
@scholzj @ppatierno I've reverted back to the default values in checkstyle.xml and in the process had to add four SuppressWarnings in total. Is this the most satisfactory way of handling it? Thanks